src/cpp/core/json/Json.cpp

```diff
@@ -245,6 +259,162 @@ bool parse(const std::string& input, Value* pValue)
    return pValue->parse(input);
 }
 
+Error parse(const std::string& input, const ErrorLocation& location, Value* pValue)
+{
+   return pValue->parse(input, location);
+}
+
+json::Object getSchemaDefaults(const Object& schema)
+{
+   json::Object result;
+   Object::iterator objType = schema.find("type");
+   if (objType == schema.end() ||
+       (*objType).value().type() != json::StringType ||
+       (*objType).value().get_str() != "object")
+   {
+      // Nothing to do for non-object types
+      return result;
+   }
+
+   Object::iterator objProperties = schema.find("properties");
+   if (objProperties == schema.end() ||
+       (*objProperties).value().type() != json::ObjectType)
+   {
+      // Nothing to do for types with no properties
+      return result;
+   }
+
+   // Iterate over all the properties specified in the schema
+   const json::Object& properties = (*objProperties).value().get_obj();
+   for (auto prop: properties)
+   {
+      // JSON schema specifies that properties are defined with objects
+      if (prop.value().type() != json::ObjectType)
+         continue;
+
+      const json::Object& definition = prop.value().get_obj();
+      Object::iterator def = definition.find("default");
+      if (def == definition.end())
+      {
+         // We didn't find a default value for this property, so recurse and see if it is an
+         // object with its own defaults.
+         json::Object child = getSchemaDefaults(definition);
+         if (!child.empty())
+         {
+            // We found defaults inside the object; use them
+            result[prop.name()] = child;
+         }
+      }
+      else
+      {
+         // Use the default
+         result[prop.name()] = (*def).value();
```

Comment:
This will actually end up moving def's value due to the ref semantics we have here, since `.value()` will generate a `json::Value&&` which will subsequently be move-assigned to `result[prop.name()]`. You should replace this will `(*def).value().clone()`